src/server/pkg/tarutil/util.go

```diff
@@ -0,0 +1,106 @@
+package tarutil
+
+import (
+	"bytes"
+	"io"
+
+	"github.com/pachyderm/pachyderm/src/client/pkg/errors"
+	"github.com/pachyderm/pachyderm/src/server/pkg/tar"
+)
+
+type File interface {
+	Header() (*tar.Header, error)
+	Content(io.Writer) error
+}
+
+type file struct {
+	hdr  *tar.Header
+	data []byte
+}
+
+func NewFile(name string, data []byte) File {
+	return newFile(NewHeader(name, int64(len(data))), data)
+}
+
+func newFile(hdr *tar.Header, data []byte) File {
+	return &file{
+		hdr:  hdr,
+		data: data,
+	}
+}
+
+func NewHeader(name string, size int64) *tar.Header {
```

Comment:
This should set the TypeFlag based on trailing slash like in the package.  And then also zero out size.

Reply:
Not sure I understand what you are suggesting here? The TypeFlag is set at write time (`WriteHeader`) and the size should be set in accordance with the size of the data (so a directory would be size zero).

Reply:
My concern is that this is a public constructor. The normal usecase is creating a header right before writing it.  This seems like it could be used to create and pass around an invalid header.

Reply:
My thinking was that it would expose a way for a client to create a default header. We either don't have a way to create a default header for a client, or we do and we run the risk that they could incorrectly make modifications to the header. I think the latter is preferable.

Reply:
Okay. This is fine.  I'm not even sure how much use it will get, since a header literal is about as simple.